### ISSUE TITLE
Can not connect to the broker

### ISSUE THREAD

**Issue by yldzmuhammed** [vernemq/vernemq#1408]
### To help us save time and help you faster

Before opening a new issue, please make sure that isn't already an open (or
closed/resolved) issue concerning the problem. If unsure if older issues are
about the exact same thing, just open a new issue rather than posting in the old
one - but please reference the possibly related issues in the new issue.

Please do not insert images of text, but insert the text instead.

### Environment

- VerneMQ Version: last version from build for ubuntu bionic
- OS: ubuntu 18
- Erlang/OTP version (if building from source):
- VerneMQ configuration (vernemq.conf) or the changes from the default 
## 
## 
##   - one of: yes, no
accept_eula = yes

## 
## 
##   - on or off
allow_anonymous = on

## 
## 
##   - on or off
allow_register_during_netsplit = off

## 
## 
##   - on or off
allow_publish_during_netsplit = off

## 
## 
##   - on or off
allow_subscribe_during_netsplit = off

## 
## 
##   - on or off
allow_unsubscribe_during_netsplit = off

## 
## 
##   - on or off
allow_multiple_sessions = off

## 
## 
##   - on or off
coordinate_registrations = on

## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
##   - an integer
max_inflight_messages = 20

## 
## 
##   - an integer
max_online_messages = 1000

## 
## 
##   - an integer
max_offline_messages = 1000

## 
## 
##   - an integer
max_message_size = 0

## 
## 
##   - on or off
upgrade_outgoing_qos = off

## 
## 
## 
##   - the text "infinity"
listener.max_connections = 10000

## 
## 
##   - an integer
listener.nr_of_acceptors = 10

## 
## 
##   - an IP/port pair, e.g. 127.0.0.1:10011
listener.tcp.default = 127.0.0.1:1883
listener.tcp.my_lstn = 0.0.0.0:1880

## 
## 
## 
## 
##   - an IP/port pair, e.g. 127.0.0.1:10011
listener.vmq.clustering = 0.0.0.0:44053

## 
## 
##   - an IP/port pair, e.g. 127.0.0.1:10011
listener.http.default = 127.0.0.1:8888
listener.http.myhttp  = 0.0.0.0:8880

## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
##   - on or off
systree_enabled = on

## 
## 
##   - an integer
systree_interval = 20000

## 
## 
##   - on or off
graphite_enabled = off

## 
## 
##   - text
graphite_host = localhost

## 
## 
##   - an integer
graphite_port = 2003

## 
## 
##   - an integer
graphite_interval = 20000

## 
## 
## 
## 
## 
## 
##   - text
shared_subscription_policy = prefer_local

## 
## 
## 
## 
## 
##   - on or off
plugins.vmq_passwd = on

## 
## 
##   - on or off
plugins.vmq_acl = on

## 
## 
##   - on or off
plugins.vmq_diversity = off

## 
## 
##   - on or off
plugins.vmq_webhooks = off

## 
## 
##   - on or off
plugins.vmq_bridge = off

## 
## 
##   - one of: vmq_plumtree, vmq_swc
metadata_plugin = vmq_plumtree

## 
## 
##   - the path to a file
vmq_acl.acl_file = /etc/vernemq/vmq.acl

## 
## 
##   - an integer
vmq_acl.acl_reload_interval = 10

## 
## 
##   - the path to a file
vmq_passwd.password_file = /etc/vernemq/vmq.passwd

## 
## 
##   - an integer
vmq_passwd.password_reload_interval = 10

## 
## 
##   - the path to a directory
vmq_diversity.script_dir = /usr/share/vernemq/lua

## 
## 
##   - on or off
vmq_diversity.auth_postgres.enabled = off

## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
##   - on or off
vmq_diversity.postgres.ssl = off

## 
## 
## 
## 
## 
## 
## 
## 
##   - one of: crypt, bcrypt
vmq_diversity.postgres.password_hash_method = crypt

## 
## 
##   - on or off
vmq_diversity.auth_cockroachdb.enabled = off

## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
##   - on or off
vmq_diversity.cockroachdb.ssl = on

## 
## 
## 
## 
## 
## 
## 
## 
##   - one of: sha256, bcrypt
vmq_diversity.cockroachdb.password_hash_method = bcrypt

## 
## 
##   - on or off
vmq_diversity.auth_mysql.enabled = off

## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
##   - one of: password, md5, sha1, sha256
vmq_diversity.mysql.password_hash_method = password

## 
## 
##   - on or off
vmq_diversity.auth_mongodb.enabled = off

## 
## 
## 
## 
## 
## 
## 
## 
## 
##   - on or off
vmq_diversity.mongodb.ssl = off

## 
## 
## 
## 
## 
## 
## 
## 
##   - on or off
vmq_diversity.auth_redis.enabled = off

## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
##   - one of: auto
vmq_bcrypt.pool_size = 1

## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
## 
##   - one of: off, file, console, both
log.console = file

## 
## 
##   - one of: debug, info, warning, error
log.console.level = info

## 
## 
##   - the path to a file
log.console.file = /var/log/vernemq/console.log

## 
## 
##   - the path to a file
log.error.file = /var/log/vernemq/error.log

## 
## 
##   - on or off
log.syslog = off

## 
## 
##   - on or off
log.crash = on

## 
## 
##   - the path to a file
log.crash.file = /var/log/vernemq/crash.log

## 
## 
##   - a byte size with units, e.g. 10GB
log.crash.maximum_message_size = 64KB

## 
## 
##   - a byte size with units, e.g. 10GB
log.crash.size = 10MB

## 
## 
##   - text
log.crash.rotation = $D0

## 
## 
##   - the text "current"
log.crash.rotation.keep = 5

## 
## 
##   - text
nodename = VerneMQ@127.0.0.1

## 
## 
##   - text
distributed_cookie = vmq

## 
## 
##   - an integer
erlang.async_threads = 64

## 
## 
##   - an integer
erlang.max_ports = 262144

## 
## 
## 
## 
## 
##   - an integer
leveldb.maximum_memory.percent = 70

- Cluster size/standalone: not sure cluster or standalone version.

### Expected behavior
- Please describe the expected behaviour.
Connect from online or any other mqtt client. 

### Actual behaviour

- Please include minimal steps to reproduce the issue, if possible.
There is no successfull connection. When i try to connect from online mqtt client, it gives me connection error. But no detail (timeout, refused or else).
When i try to connect to my server from tcp socket client, it connects to the socket 1880 and after about 3 seconds, close the connection.

What do i miss? Is my config file is wrong?

I tried with allow_ananymous on and off. Still no luck.

**Comment by ioolkos** [comment]
@yldzmuhammed try configuring the MQTT listener using the specific IP, instead of 0.0.0.0
We've sometimes seen issues with that, especially on AWS and/or Docker

**Comment by yldzmuhammed** [comment]
Dear @ioolkos can you explain me taht what is tcp listener exactly? I was thinking as a specific io address to listen. And If I wantt to listen every io address, it should be 0.0.0.0. Do I missing some thing? 
Is this config related with cluster? What if I don't want to use cluster and use standalone? 
I putted my config file. Can you tell me what should I do? 
Great thanks.

**Comment by yldzmuhammed** [comment]
Can some one explain how can i up and run this app?

I want to use it for simple usage for now.

**Comment by ioolkos** [comment]
@yldzmuhammed have you tried what I suggested?

**Comment by yldzmuhammed** [comment]
@ioolkos if would understand what you said i would. but i did not. Sorry. Can you explain it?

**Comment by yldzmuhammed** [comment]
What do you mean by specific ip. Is it client's ip? I am not using cluster. Just standalone.

**Comment by ioolkos** [comment]
The Server IP, not 0.0.0.0
Configure the MQTT listener to that in your vernemq.conf file

**Comment by yldzmuhammed** [comment]
@ioolkos i tried. But no luck. When i try with pyhton code, it responses with connection refused.

**Comment by yldzmuhammed** [comment]
Amazon aws servers has 2 ip address. One public ip, other private ip. I tried with both.

BTW, these configs can bu made on the fly, correct?

**Comment by yldzmuhammed** [comment]
What exactly is this tcp listener? What does it do?I am a little confused.

**Comment by yldzmuhammed** [comment]
Is there any body?

**Comment by larshesel** [comment]
https://docs.vernemq.com/configuration/listeners

**Comment by KhairulM** [comment]
Just for anyone still struggling with this. My setup was:

VerneMQ version 1.13.0
Running on AWS EC2 with a security group allowing TCP access to port 1883 from 0.0.0.0/0 (basically anywhere)
Had `allow_anonymous = on` and `accept_eula = yes`
Confirmed that VerneMQ is running through `systemctl status vernemq.service`
Confirmed that the EC2 instance is accessible from the public internet through SSH

What I tried was:
Connect to VerneMQ using local MQTT client by the EC2 public DNS name

What I got was:
Connection refused on multiple MQTT clients that I tried

What I did was:
1. Changed this line `listener.tcp.name = 127.0.0.1:1883` to `listener.tcp.default  = <private IP of the EC2>:1883`
2. Restart VerneMQ through `sudo systemctl restart vernemq.service`

Now I can connect from my local MQTT client
